bits.go

```diff
@@ -266,3 +267,50 @@ func fromDeltas(data []byte, buf []uint32) []uint32 {
 	}
 	return buf
 }
+
+// condenseRuneOffsets converts the mostly-predictable runeOffset input
+// into a shorter form based on tuples of the unexpected values.
+// The input is roughly a sequence of y values [100*i for i in range(n)],
+// so we just store (x, y) points where the expectation is violated.
+//
+// This is very rarely a slight pessimization, on repos where there are frequent
+// non-ASCII characters.
+func condenseRuneOffsets(off []uint32) []uint32 {
```

Comment:
Not sure why we need to copy `tmp` into `out`? Seems like we can append to out directly in the loop.

```suggestion
func condenseRuneOffsets(off []uint32) (out []uint32) {
```

Reply:
added comment:
`// copy the slice to ensure it doesn't waste unused capacity`